client.go

```diff
@@ -92,6 +99,7 @@ func NewClient(url, usr, pwd string, mods ...func(*Client)) (Client, error) {
 		BackoffDelayFactor:  DefaultBackoffDelayFactor,
 		AuthenticationMutex: &sync.Mutex{},
 		RateLimiterBucket:   ratelimit.NewBucketWithRate(1.66, 1), // 1.66 req/s == 100 req/min
+		writingMutex:        &sync.Mutex{},
```

Comment:
cosmetic -> `WritingMutex`

Reply:
Making it public adds risk. Do you see any potential benefit whatsoever? If anybody Locks it, they can either Unlock, Get from API, or sleep?